tests/Session/SessionTest.php

```diff
@@ -0,0 +1,138 @@
+    {
+        $session = new Session();
+        $session->set('get', 'set');
+        self::assertEquals('set', $session->get('get'));
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testHas()
+    {
+        $session = new Session();
+        $session->set('has', 'has');
+        self::assertTrue($session->has('has'));
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testClose()
+    {
+        $session = new Session();
+        $session->set('close', 'close');
+        $session->close();
+        self::assertEquals(PHP_SESSION_NONE, session_status());
+        // because session_destroy() in tearDown doesn't work as expected
+        // we need to open session and then destroy it
+        $session->open();
+        $session->destroy();
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testRegenerateID()
+    {
+        $session = new Session();
+        $session->open();
+        $id = $session->getId();
+        $session->regenerateId();
+        self::assertNotEquals($id, $session->getId());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testDiscard()
+    {
+        $session = new Session();
+        $session->set('discard', 'discard');
+        $session->discard();
+        self::assertEmpty($session->get('discard'));
+    }
+
+    public function testGetName()
+    {
+        $session = new Session();
+        self::assertEquals($session->getName(), session_name());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testPull()
+    {
+        $session = new Session();
+        $session->set('pull', 'pull');
+        self::assertEquals('pull', $session->pull('pull'));
+        self::assertEmpty($session->get('pull'));
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testAll()
+    {
+        $session = new Session();
+        $session->set('1', 1);
+        $session->set('2', 2);
+        self::assertEquals(['1' => 1, '2' => 2], $session->all());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testClear()
+    {
+        $session = new Session();
+        $session->set('1', 1);
+        $session->clear();
+        self::assertEmpty($session->all());
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testSetId()
+    {
+        $session = new Session();
+        $session->setId('sessionId');
+        $session->open();
+        self::assertEquals(session_id(), $session->getId());
+    }
+
+    public function testGetCookieParameters()
+    {
+        $session = new Session();
+        self::assertEquals(session_get_cookie_params(), $session->getCookieParameters());
+    }
+
+    /**
+     * @expectedException \RuntimeException
+     */
+    public function testAlreadyStartedException()
+    {
+        $session = new Session();
+        $session->set('1', 1);
+        $session = new Session();
```

Comment:
```suggestion
        new Session();
```

Reply:
We can slightly simple (with `$session->open()`).